kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/operator/KroxyliciousOperatorBundleInstaller.java

```diff
@@ -0,0 +1,483 @@
+        this.bindingsNamespaces = builder.bindingsNamespaces;
+        this.operationTimeout = builder.operationTimeout;
+        this.reconciliationInterval = builder.reconciliationInterval;
+        this.extraLabels = builder.extraLabels;
+        this.replicas = builder.replicas;
+
+        // assign defaults is something is not specified
+        if (this.kroxyliciousOperatorName == null || this.kroxyliciousOperatorName.isEmpty()) {
+            this.kroxyliciousOperatorName = Constants.KO_DEPLOYMENT_NAME;
+        }
+        // if namespace is not set we install operator to 'kroxylicious-operator'
+        if (this.namespaceInstallTo == null || this.namespaceInstallTo.isEmpty()) {
+            this.namespaceInstallTo = Constants.KO_NAMESPACE;
+        }
+        if (this.namespaceToWatch == null) {
+            this.namespaceToWatch = this.namespaceInstallTo;
+        }
+        if (this.bindingsNamespaces == null) {
+            this.bindingsNamespaces = new ArrayList<>();
+            this.bindingsNamespaces.add(this.namespaceInstallTo);
+        }
+        if (this.operationTimeout == null) {
+            this.operationTimeout = Constants.KO_OPERATION_TIMEOUT_DEFAULT;
+        }
+        if (this.reconciliationInterval == null) {
+            this.reconciliationInterval = Constants.RECONCILIATION_INTERVAL;
+        }
+        if (this.extraLabels == null) {
+            this.extraLabels = new HashMap<>();
+        }
+    }
+
+    private boolean isKroxyliciousOperatorNamespaceNotCreated() {
+        return extensionContext == null
+                || extensionContext.getStore(ExtensionContext.Namespace.GLOBAL).get(Constants.PREPARE_OPERATOR_ENV_KEY + namespaceInstallTo) == null;
+    }
+
+    private void createKroxyliciousOperatorNamespaceIfPossible() {
+        if (isKroxyliciousOperatorNamespaceNotCreated()) {
+            cluster.setNamespace(namespaceInstallTo);
+
+            if (this.extensionContext != null) {
+                ResourceManager.getStoredResources().computeIfAbsent(this.extensionContext.getDisplayName(), k -> new Stack<>());
+                ResourceManager.getStoredResources().get(this.extensionContext.getDisplayName()).push(
+                        new ResourceItem<>(this::deleteKroxyliciousOperatorNamespace));
+
+                NamespaceUtils.createNamespaces(namespaceInstallTo, bindingsNamespaces);
+
+                this.extensionContext.getStore(ExtensionContext.Namespace.GLOBAL).put(Constants.PREPARE_OPERATOR_ENV_KEY + namespaceInstallTo, true);
+            }
+        }
+    }
+
+    /**
+     * Perform application of ServiceAccount, Roles and CRDs needed for proper cluster operator deployment.
+     * Configuration files are loaded from kroxylicious-operator directory.
+     */
+    public void applyClusterOperatorInstallFiles(String namespaceName) {
+        List<File> operatorFiles = Arrays.stream(Objects.requireNonNull(new File(Constants.PATH_TO_OPERATOR_INSTALL_FILES).listFiles())).sorted()
```

Comment:
Fabric8 lets you do the equivalent of an kubectl apply.   You should apply the CRDs first, then the rest of the resources.   You don't need all this case by case handling.

https://github.com/fabric8io/kubernetes-client/blob/main/kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateOrReplaceResourceList.java